### ISSUE TITLE
Bank account synchronization with STET PSD2

### ISSUE THREAD

**Issue by azmeuk** [nlogozzo/NickvisionMoney#170]
[PSD2 is an European directive](https://en.wikipedia.org/wiki/Payment_Services_Directive#Revised_Directive_on_Payment_Services_(PSD2)) that promotes - among other things - open banking. 
The [STET PSD2 API](https://www.stet.eu/en/psd2/) describes several operations that can be realized on user bank accounts, and among others there is AISP (Account Information Service Providers) that allows access to users account transactions. It is reputed to be a de facto standard in Europe, and is for instance implemented in most French banks: [CIC](https://www.cic.fr/oauth2/en/devportal/stetpsd2-spec-v2.1.html), [Crédit Agricole](https://www.ca-cib.fr/nos-expertises/international-trade-transaction-banking/transaction-banking/plateforme-api)- [La Banque Postale](https://developer.labanquepostale.com/), the [BPCE Group](https://www.api.89c3.com) (Banque Populaire, Caisse d'Épargne, Crédit Coopératif) etc.

I know this is more an epic than a quick task, but I would suggest implementing a mechanism to synchronize bank account transactions in money using this standard.

**Comment by darkdragon-001** [comment]
In Germany, mostly [FinTS](https://en.m.wikipedia.org/wiki/FinTS) is used (replacing [HBCI](https://de.m.wikipedia.org/wiki/Homebanking_Computer_Interface)).

**Comment by darkdragon-001** [comment]
I just noticed that [GNUCash uses](https://wiki.gnucash.org/wiki/De/HBCI) the [AQBanking library](https://github.com/aqbanking/aqbanking#readme) for import/export from/to various formats and online APIs.

**Comment by gunwald** [comment]
Just for the record: There is also an applications called [»Banking«](https://gitlab.com/tabos/banking) which works as a FinTS client:

> This is a FinTS banking application using python-fints as backend library. This application is able to connect to your bank and will show your current balance and transactions. It has been created for my upcoming Librem 5 smartphone by Purism.

I never got it working, but maybe you could combine efforts.

**Comment by jbingram** [comment]
@gunwald Yep, but [»Banking«](https://gitlab.com/tabos/banking) only works with German banks, and the maintainer is a busy guy, has no time, and thus no plan to expand to other foreign bank transfer protocols.

**Comment by ayykamp** [comment]
This would be an amazing feature, being able to automatically import all transactions and then be able to categorize them in Denaro.

The project Firefly-III uses [proprietary APIs](https://docs.firefly-iii.org/data-importer/faq/spectre-and-nordigen/) to do this but an open source implementation would of course be ideal :)

**Comment by ayykamp** [comment]
See also https://gitlab.com/tabos/saldo/-/issues/62

**Comment by xplosionmind** [comment]
Hi! I am very much interested in this topic, even though:

- I am not a developer
- I am completely new to this topic/area

My bank is the Italian [Banca Etica](https://bancaetica.it). How can I verify what protocol and what systems they use? If I wanted to open a request to implement an open banking standard, what could I do?

Thanks a lot!

Best,  
Tommi

**Comment by Kuchenmampfer** [comment]
I already use [saldo](https://gitlab.com/tabos/saldo/-/issues/62) and currently just copy everything from there into denaro by hand. So for me, any way of automatically retrieving transactions from saldo would be enough. However, this would of course only cover german banks until saldo expands
